infra/Pos/Slotting/MemState/Class.hs

```diff
@@ -8,24 +8,52 @@ import           Control.Monad.Trans (MonadTrans)
 import           Universum
 
 import           Pos.Core.Types      (EpochIndex, Timestamp)
-import           Pos.Slotting.Types  (SlottingData)
+import           Pos.Slotting.Types  (EpochSlottingData)
+
 
 -- | 'MonadSlotsData' provides access to data necessary for slotting to work.
+-- Both _current_ and _next_ epoch @SlottingData@ need to be present since it's impossible
+-- to create @SlottingData@ without them.
 class Monad m => MonadSlotsData m where
 
-    getSystemStart :: m Timestamp
+    getSystemStartM :: m Timestamp
+
+    getAllEpochIndicesM :: m [EpochIndex]
```

Comment:
I didn't get why this function is here.
AFAIU it's just `[0..nextEpochIndex]`, isn't it?